src/Client.php

```diff
@@ -542,21 +549,32 @@ protected function send(array $data, array $tags = [])
      */
     protected function sendMessages(array $messages)
     {
-        $socket = @fsockopen('udp://' . $this->host, $this->port, $errno, $errstr, $this->timeout);
-        if (!$socket) {
-            if ($this->throwExceptions) {
-                throw new ConnectionException($this, '(' . $errno . ') ' . $errstr);
-            } else {
-                trigger_error(
-                    sprintf('StatsD server connection failed (udp://%s:%d)', $this->host, $this->port),
-                    E_USER_WARNING
-                );
-                return $this;
+        if($this->socket === null) {
+            $this->socket = @fsockopen('udp://' . $this->host, $this->port, $errno, $errstr, $this->timeout);
+            if (!$this->socket) {
+                if ($this->throwExceptions) {
+                    throw new ConnectionException($this, '(' . $errno . ') ' . $errstr);
+                } else {
+                    trigger_error(
+                        sprintf('StatsD server connection failed (udp://%s:%d)', $this->host, $this->port),
+                        E_USER_WARNING
+                    );
+                }
             }
         }
-        $this->message = implode("\n", $messages);
-        @fwrite($socket, $this->message);
-        fclose($socket);
+        
+        if($this->socket) {
+            $this->message = implode("\n", $messages);
+            @fwrite($socket, $this->message);
```

Comment:
@h-bragg oh, right. sorry, my fault. I just updated code directly on GitHub. my local version is completely different with some another fixes: 
1. because connection timeout is not enough. I've got some problems because of this already, so here is a fix:
```
        if($this->socket) {
            $this->message = implode("\n", $messages);
            $oldTimeout = ini_set('stream_set_timeout', $this->timeout);
            @fwrite($this->socket, $this->message);
            ini_set('stream_set_timeout', $oldTimeout);
        }
```
2. because when trigger_error in PHP 7 it throws exception, so you shouldn't use trigger_error
```
        if($this->socket === null) {
            $this->socket = @fsockopen('udp://' . $this->host, $this->port, $errno, $errstr, $this->timeout);
            if(!$this->socket) {
                $this->socket = false;

                if($this->throwExceptions) {
                    throw new ConnectionException($this, '(' . $errno . ') ' . $errstr);
                }

                return $this;
            }
        }
```